Automattic/wp-calypso
# Jetpack Backup: Give a checkbox to save staging credentials on the Copy to staging screen

## Proposed Changes

* Give a `Save for later` checkbox on Jetpack Cloud's _Copy to Staging_ credentials screen
* Record a tracks event `calypso_jetpack_advanced_credentials_flow_credentials_save_staging` when `Save for later` checkbox is checked and the form is submitted.

| Before  | After |
| ------------- | ------------- |
| <img width="708" alt="Screenshot 2023-10-20 at 6 17 40 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/d8c3fe21-20d8-496e-8072-7a0df8edfa34">  | <img width="715" alt="Screenshot 2023-10-20 at 6 17 54 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/138ac26a-3867-400c-a388-5ec7ddd11296">  |

| Unchecked  | Checked |
| ------------- | ------------- |
| <img width="376" alt="Screenshot 2023-10-20 at 6 51 44 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/bc5b97bb-3c47-4961-80a3-2a77a525eaac"> | <img width="359" alt="Screenshot 2023-10-20 at 6 51 50 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/8e7617d7-32c1-4f51-9dc5-9da3918f831b"> |


## Testing Instructions

- Spin up Calypso Blue and Jetpack Cloud live branches
- Visit a site's activity log and click on _Copy this site_ button
- On _Set a destination site_ screen, click on _Enter credentials for a new destination site +_ link
  
  <img width="395" alt="Screenshot 2023-10-20 at 6 56 14 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/73b90648-6bb9-49a9-915f-24b988a40544">

- Ensure the form has a _Save for later_ checkbox just above _Confirm credentials_ button.
   
   <img width="376" alt="Screenshot 2023-10-20 at 6 51 44 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/bc5b97bb-3c47-4961-80a3-2a77a525eaac">

- Fill in the SSH/SFTP credentials of a non-Jetpack-connected site. Before submitting check the `Save for later` checkbox, the submit button text will change to `Test and save credentials` 

  <img width="359" alt="Screenshot 2023-10-20 at 6 51 50 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/8e7617d7-32c1-4f51-9dc5-9da3918f831b">
- Open Developer tools to verify the tracks event on submit, verify only if the `Save for later` checkbox is checked and the form is submitted - the tracks event `calypso_jetpack_advanced_credentials_flow_credentials_save_staging` should get recorded.
  
   <img width="727" alt="Screenshot 2023-10-20 at 7 04 54 PM" src="https://github.com/Automattic/wp-calypso/assets/13975226/c6bc3688-70ee-40a0-9b43-a77240c9f40d">

- Additionally, verify if the credentials are saved with `staging-<identifier>` role in RWDB's credentials section of this Jetpack site.

@grbicsanja / @keoshi or anyone - Please check if the text for the `Save for later` checkbox/toggling of the submit button text on change of checkbox state change is acceptable or can be improved/simplified :)

## Pre-merge Checklist
